src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCorePackager.java

```diff
@@ -201,13 +206,11 @@ public static List<FileInfo> parseDatalink(URL url, DownloadRequest request, Map
                         String extension = "unknown";
                         extension = content_type != null ? getExtFromURL(content_type) : getExtFromURL(productUrl);
                         ext_file_name += "." + extension;
-                        ext_file_name = ("folder".equalsIgnoreCase(zipType) || (zipType == null && makeDataLinkFolder))
-                                ? prepend_file_name + "/" + ext_file_name
-                                : ext_file_name;
+                        ext_file_name = ("flat".equalsIgnoreCase(zipType) || (!makeDataLinkFolder)) ?
+                                ext_file_name : prepend_file_name + "/" + ext_file_name;
```

Comment:
Yes, here! You are in-fact just doing a binary operation on zipType=='flat', 'folder' doesn't really have any use in your server-side parser. 

This will then become: 
```suggestion
                        ext_file_name = (isFlattenedStructure || (!makeDataLinkFolder)) ?
                                ext_file_name : prepend_file_name + "/" + ext_file_name;
```

Just make sure to replace
`String zipType = request.getParam(ZIP_TYPE);` with `boolean isFlattenedStructure = getBooleanParam('fieldKeyYouDefinedAtClientSide');`